services/fileuploader/fileuploader.go

```diff
@@ -67,20 +82,20 @@ func (p *provider) getStorageSettings(workspaceID string) (StorageSettings, erro
 	defer p.mu.RUnlock()
 	settings, ok := p.storageSettings[workspaceID]
 	if !ok {
-		return StorageSettings{}, NoStorageForWorkspaceError
+		return StorageSettings{}, ErrNoStorageForWorkspace
 	}
 	return settings, nil
 }
 
 func (p *provider) GetFileManager(workspaceID string) (filemanager.FileManager, error) {
```

Comment:
We should consider introducing a context here and do a `select` with `<-p.init`.

Reply:
not using `ctx` just to keep the signature same.
But adopting the select and returning another sentinel error